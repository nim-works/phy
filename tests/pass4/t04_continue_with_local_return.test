discard """
  description: "
    Ensure the expression in a `Continue` targeting the exit is processed
    properly (i.e., that the locals within use the correct IDs)
  "
"""
(TypeDefs
  (Int 8)
  (ProcTy (Type 0))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params) (Locals)
      (Continue 1 (IntVal 100) (List))
    )
    (Continuation (Params (Type 0)) (Locals)
      (Continue 2 (IntVal 200)
        (List (Move (Local 0))))
    )
    (Continuation (Params (Type 0) (Type 0)) (Locals)
      (Continue 3
        (Add (Type 0)
          (Copy (Local 0))
          (Copy (Local 1)))
        (List))
    )
    (Continuation (Params (Type 0))))
  )
)
