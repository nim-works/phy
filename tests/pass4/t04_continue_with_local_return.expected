;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Type 0)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0) (Type 0))
    (Continuations
      (Continuation (Params)
        (Locals (Local 0))
        (Asgn (Local 0) (IntVal 100))
        (Continue 1))
      (Continuation (Params)
        (Locals (Local 0) (Local 1))
        (Asgn (Local 1) (IntVal 200))
        (Continue 2))
      (Continuation (Params)
        (Locals (Local 1) (Local 0))
        (Continue 3
          (Add (Type 0)
            (Copy (Local 1))
            (Copy (Local 0)))))
      (Continuation
        (Params (Type 0))))))
